source/agora/node/Ledger.d

```diff
@@ -1145,26 +1182,27 @@ public class ValidatingLedger : Ledger
                 return err;
             };
 
-            if (auto reason = tx.isInvalidReason(this.engine, utxo_finder,
-                next_height, checkAndAcc))
+            if (auto reason = tx.isInvalidReason(
+                    this.engine, utxo_finder, height, checkAndAcc))
                 log.trace("Rejected invalid ('{}') tx: {}", reason, tx);
             else
-                data.tx_set ~= hash;
+                result ~= hash;
 
-            if (data.tx_set.length >= max_txs)
+            if (result.length >= max_txs)
             {
-                data.tx_set.sort();
-                return;
+                result.sort();
+                return result;
```

Comment:
Why not `return result.sort();` then becomes 3 lines less?

Reply:
Because the return value is a `SortedRange`.